### PR TITLE
CI: Create GitHub Actions workflow that tests LoopGPT

### PR DESCRIPTION
**Pull request by EwoutH**
Adds a continuous integration configuration which uses GitHub Actions to run the LoopGPT [tests](https://github.com/farizrahman4u/loopgpt/tree/main/tests) using Pytest. It runs on each push and pull request, and also once a week on Monday and when manually triggered.

GitHub Actions needs to be enabled at https://github.com/farizrahman4u/loopgpt/actions

### CODE REVIEW

**Comment by FayazRahman** [comment]
@EwoutH Thanks a lot for the PR! We'll get this merged soon.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  schedule:
+    - cron: '0 6 * * 1'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (farizrahman4u) [inline .github/workflows/test.yml:15]:
```suggestion
        os: [ubuntu-latest]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        python-version: ["3.11"]
```
Comment (farizrahman4u) [inline .github/workflows/test.yml:16]:
```suggestion
        python-version: ["3.10"]
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        include:
+          - os: ubuntu-latest
+            python-version: "3.10"
+          - os: ubuntu-latest
+            python-version: "3.9"
```
Comment (farizrahman4u) [inline .github/workflows/test.yml:21]:
```suggestion
```

**Review by farizrahman4u — APPROVED** [review]

**Comment by farizrahman4u** [comment]
Thank you!

**Pull request merged**